### DISCUSSION TITLE
OG image generation doesn't work with Static Export

### DISCUSSION THREAD

**Discussion by lr0pb** [vercel/next.js#55890] [App Router]
OG image generation [docs](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#generate-images-using-code-js-ts-tsx) saying that
> By default, generated images are [statically optimized](https://nextjs.org/docs/app/building-your-application/rendering/server-components#static-rendering-default) (generated at build time and cached) unless they use [dynamic functions](https://nextjs.org/docs/app/building-your-application/rendering/server-components#server-rendering-strategies#dynamic-functions) or uncached data.

Reading this I assumed, that OG image generation also works for static export using `output: export` config option.
However no information about this as either supported or unsupported feature in the static export [docs](https://nextjs.org/docs/app/building-your-application/deploying/static-exports) and as I tested - in practice it works to be used with the server.

It would be great to clarify this question in docs and even greater to make it possible to use in the static export, because this feature is essential for both dynamic and static sites.

**Comment by jsbisht** [comment]
Related to https://github.com/vercel/next.js/issues/51147
